components/core/src/clp/string_utils/string_utils.hpp

```diff
@@ -61,6 +61,16 @@ std::string replace_characters(
  */
 void to_lower(std::string& str);
 
+/**
+ * Replace unescaped instances of src_char with target_char in src_string
+ * @param escape_char The character which indicates escape
+ * @param src_char The character which should be replaced if unescaped
+ * @param target_char The character which should replace src_char
+ * @param search_string The string which needs to have src_char replaced
```

Comment:
I think we can remove the comments for src_char, target_char and search_string.
Also don't forget to rename search_string to src_string in the comment.